tempesta_fw/http.c

```diff
@@ -683,12 +740,37 @@ tfw_http_req_process(TfwConnection *conn, struct sk_buff *skb, unsigned int off)
 			}
 		}
 		/*
+		 * Complete HTTP message has been collected and successfully
+		 * processed. Mark the message as complete in @conn, because
+		 * further handling of @conn depends on that. Future SKBs
+		 * will be put in a new message.
+		 * Otherwise, the function returns from inside the loop.
+		 * @conn->msg holds the reference to the message, which can
+		 * be used to release it.
+		 */
+		conn->msg = NULL;
```

Comment:
Why do we need the change? Initially it was done to properly allocate a new message in ```tfw_connection_recv()```. While ```tfw_http_req_process()``` is called only from ```tfw_connection_recv()```, when the message is already allocated. Since you're assigns ```hmsib``` to ```conn->msg``` few lines below, it seems you need the change for ```tfw_cache_req_process((TfwHttpReq *)hmreq, tfw_http_req_cache_cb)```. And this isn't clear for me: both the function don't do anything with ```conn->msg``` and it should be properly freed by ```tfw_http_conn_msg_unlink()``` on connection closing...